File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    
    global count
    count =0
    
    
    
    def dfs(level,n,sum):
        global count
        
        
        if level==len(numbers):
            if sum ==target:
                count+=1
            return
        
        dfs(level+1,n+1,sum+numbers[n])
        dfs(level+1,n+1,sum-numbers[n])
        return
        
            
    dfs(0,0,0)

    return count

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    function dfs(level, sum) {
        if (level === numbers.length) {
            return sum === target ? 1 : 0;
        }

        const add = dfs(level + 1, sum + numbers[level]);
        const subtract = dfs(level + 1, sum - numbers[level]);

        return add + subtract;
    }
    
    return dfs(0, 0);
}
